azext_edge/edge/providers/check/opcua.py

```diff
@@ -0,0 +1,247 @@
+        )
+        check_manager.add_display(target_name=target_asset_types, display=Padding(fetch_asset_types_error_text, (0, 0, 0, 8)))
+
+    for (namespace, asset_types) in resources_grouped_by_namespace(all_asset_types):
+        check_manager.add_target(target_name=target_asset_types, namespace=namespace, conditions=asset_type_conditions)
+        check_manager.add_display(
+            target_name=target_asset_types,
+            namespace=namespace,
+            display=Padding(
+                f"OPCUA asset types in namespace {{[purple]{namespace}[/purple]}}",
+                (0, 0, 0, 8)
+            )
+        )
+
+        asset_types: List[dict] = list(asset_types)
+        asset_types_count = len(asset_types)
+        asset_types_count_text = "- Expecting [bright_blue]>=1[/bright_blue] instance resource per namespace. {}."
+
+        if asset_types_count >= 1:
+            asset_types_count_text = asset_types_count_text.format(f"[green]Detected {asset_types_count}[/green]")
+        else:
+            asset_types_count_text = asset_types_count_text.format(f"[red]Detected {asset_types_count}[/red]")
+            check_manager.set_target_status(target_name=all_asset_types, status=CheckTaskStatus.error.value)
+        check_manager.add_display(
+            target_name=target_asset_types,
+            namespace=namespace,
+            display=Padding(asset_types_count_text, (0, 0, 0, 10))
+        )
+
+        for asset_type in asset_types:
+            asset_type_name = asset_type["metadata"]["name"]
+
+            asset_type_text = (
+                f"- Opcua asset type {{[bright_blue]{asset_type_name}[/bright_blue]}} detected."
+            )
+
+            check_manager.add_display(
+                target_name=target_asset_types,
+                namespace=namespace,
+                display=Padding(asset_type_text, (0, 0, 0, 12))
+            )
+
+            spec = asset_type["spec"]
+            if detail_level >= ResourceOutputDetailLevel.detail.value:
+                # label summarize
+                labels = spec["labels"]
+                check_manager.add_display(
+                    target_name=target_asset_types,
+                    namespace=namespace,
+                    display=Padding(
+                        f"Detected [cyan]{len(labels)}[/cyan] labels",
+                        (0, 0, 0, 16),
+                    ),
+                )
+
+                if detail_level == ResourceOutputDetailLevel.verbose.value:
+                    # remove repeated labels
+                    non_repeated_labels = list(set(labels))
+
+                    if len(non_repeated_labels) > 0:
+                        check_manager.add_display(
+                            target_name=target_asset_types,
+                            namespace=namespace,
+                            display=Padding(
+                                "[yellow](Only non repeatative labels will be displayed)[/yellow]",
```

Comment:
Is this 'Only non-repetitive labels ...' (note spelling error) display needed?  The prior display can say "Detected X **unique** labels" to simplify.

Reply:
yeah good idea! will update!